config/src/main/scala-2.10/bloop/config/ConfigEncoderDecoders.scala

```diff
@@ -109,40 +109,39 @@ object ConfigEncoderDecoders {
       case Platform.Jvm(config, mainClass) =>
         val configJson = jvmEncoder(config)
         val mainClassJson = implicitly[RootEncoder[Option[String]]].apply(mainClass)
-        Json.fromFields(List(
-          (N, Json.fromString(Platform.Jvm.name)),
-          (C, configJson),
-          (M, mainClassJson)))
+        Json.fromFields(
+          List((N, Json.fromString(Platform.Jvm.name)), (C, configJson), (M, mainClassJson)))
       case Platform.Js(config, mainClass) =>
         val configJson = jsEncoder(config)
         val mainClassJson = implicitly[RootEncoder[Option[String]]].apply(mainClass)
-        Json.fromFields(List(
-          (N, Json.fromString(Platform.Js.name)),
-          (C, configJson),
-          (M, mainClassJson)))
+        Json.fromFields(
+          List((N, Json.fromString(Platform.Js.name)), (C, configJson), (M, mainClassJson)))
       case Platform.Native(config, mainClass) =>
         val configJson = nativeEncoder(config)
         val mainClassJson = implicitly[RootEncoder[Option[String]]].apply(mainClass)
-        Json.fromFields(List(
-          (N, Json.fromString(Platform.Native.name)),
-          (C, configJson),
-          (M, mainClassJson)))
+        Json.fromFields(
+          List((N, Json.fromString(Platform.Native.name)), (C, configJson), (M, mainClassJson)))
     }
   }
 
   implicit val platformDecoder: Decoder[Platform] = new Decoder[Platform] {
-    private final val C = "config"
     override def apply(c: HCursor): Result[Platform] = {
       c.downField(N).as[String].flatMap {
-        case Platform.Jvm.name => c.get[JvmConfig](C).flatMap(config =>
-          c.get[List[String]](M).map(mainClass =>
-            Platform.Jvm(config, mainClass.headOption)))
-        case Platform.Js.name => c.get[JsConfig](C).flatMap(config =>
-          c.get[List[String]](M).map(mainClass =>
-            Platform.Js(config, mainClass.headOption)))
-        case Platform.Native.name => c.get[NativeConfig](C).flatMap(config =>
-          c.get[List[String]](M).map(mainClass =>
-            Platform.Native(config, mainClass.headOption)))
+        case Platform.Jvm.name =>
+          for {
```

Comment:
I had to write this with a for-comprehension because scalafmt was making the previous statement too big, and this way was simpler and easier to parse to my eye.

Reply:
This looks much better. Thanks.